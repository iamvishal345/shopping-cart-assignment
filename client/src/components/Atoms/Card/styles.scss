.card-container {
  display: flex;
  justify-content: space-between;
  margin: auto;
  margin-top: 2rem;
  background: white;
  position: relative;

  padding-bottom: 1rem;

  &:nth-child(2n + 1) {
    flex-direction: row-reverse;
  }
  &:after {
    content: "";
    position: absolute;
    z-index: -1;
    bottom: 10px;
    left: 10px;
    width: 98%;
    height: 15%;
    border-radius: 50%;
    box-shadow: 0 15px 10px rgb(0 0 0 / 70%);
  }
  .image-container {
    flex: 1;
    margin: auto;
  }
  .details-container {
    text-align: center;
    padding-top: 1rem;
    flex: 2;
    p {
      font-size: 12px;
    }
    .btn {
      width: fit-content;
      padding: 0.8rem 1rem;
      margin-top: 1rem;
      display: inline-block;
      text-decoration: none;
    }
  }
}
