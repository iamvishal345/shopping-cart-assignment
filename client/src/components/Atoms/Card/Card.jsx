import React from "react";
import { Link } from "react-router-dom";
import "./styles.scss";

export const Card = ({ item }) => {
  return (
    <div className="card-container">
      <div className="image-container">
        <img
          loading="lazy"
          width="100%"
          height="auto"
          src={item.imageUrl}
          alt={item.name}
        />
      </div>
      <div className="details-container">
        <h3>{item.name}</h3>
        <p>{item.description}</p>
        <Link to="/products" className="btn">
          Explore {item.key}
        </Link>
      </div>
    </div>
  );
};
