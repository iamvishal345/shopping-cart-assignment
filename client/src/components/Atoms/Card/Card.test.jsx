import React from "react";
import { shallow } from "enzyme";
import { Link } from "react-router-dom";
import { Card } from "./Card";

const item = {
  name: "testItem",
  imageUrl: "/static/images/logo.png",
  description: "Test Item Description",
  key: "test-item",
};

let wrapped = shallow(<Card item={item} />);

describe("Card", () => {
  it("should match the snapshot", () => {
    expect(wrapped).toMatchSnapshot();
  });

  it("should render the card with item details", () => {
    expect(wrapped.find("h3").text()).toEqual(item.name);
    expect(wrapped.find("p").text()).toEqual(item.description);
    expect(wrapped.find(Link).text()).toEqual(`Explore ${item.key}`);
  });
});
