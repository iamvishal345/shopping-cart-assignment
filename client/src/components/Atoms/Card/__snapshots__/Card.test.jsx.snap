// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<div
  className="card-container"
>
  <div
    className="image-container"
  >
    <img
      alt="testItem"
      height="auto"
      loading="lazy"
      src="/static/images/logo.png"
      width="100%"
    />
  </div>
  <div
    className="details-container"
  >
    <h3>
      testItem
    </h3>
    <p>
      Test Item Description
    </p>
    <Link
      className="btn"
      to="/products"
    >
      Explore 
      test-item
    </Link>
  </div>
</div>
`;
